src/hotspot/share/runtime/vmThread.cpp

```diff
@@ -401,17 +401,17 @@ void VMThread::inner_execute(VM_Operation* op) {
   HandleMark hm(VMThread::vm_thread());
 
   const char* const cause = op->cause();
-  EventMarkVMOperation em("Executing %sVM operation: %s%s%s%s",
-      prev_vm_operation != nullptr ? "nested " : "",
-      op->name(),
-      cause != nullptr ? " (" : "",
-      cause != nullptr ? cause : "",
-      cause != nullptr ? ")" : "");
-
-  log_debug(vmthread)("Evaluating %s %s VM operation: %s",
-                       prev_vm_operation != nullptr ? "nested" : "",
-                      _cur_vm_operation->evaluate_at_safepoint() ? "safepoint" : "non-safepoint",
-                      _cur_vm_operation->name());
+  stringStream ss;
+  ss.print("Executing%s%s VM operation: %s%s%s%s",
+           prev_vm_operation != nullptr ? " nested" : "",
+           op->evaluate_at_safepoint() ? " safepoint" : " non-safepoint",
+           op->name(),
+           cause != nullptr ? " (" : "",
+           cause != nullptr ? cause : "",
+           cause != nullptr ? ")" : "");
```

Comment:
Might be a little cleaner to use a separate print statement only if cause is not null.

Reply:
Yes! See new commit.